ipa-core/src/ff/boolean.rs

```diff
@@ -30,6 +31,28 @@ impl Boolean {
     }
 }
 
+impl ArrayAccess for Boolean {
+    type Output = Boolean;
+    type Iter<'a> = Once<Boolean>;
+
+    fn get(&self, index: usize) -> Option<Self::Output> {
+        if index < 1 {
+            Some(*self)
+        } else {
+            None
+        }
+    }
+
+    fn set(&mut self, index: usize, e: Self::Output) {
+        debug_assert!(index < 1);
```

Comment:
In isolation, a debug_assert seems reasonable here. On the other hand, out-of-bounds array indexing will normally panic even in release builds. I'm not sure which way makes more sense here.